src/server/game/Entities/AreaTrigger/AreaTrigger.cpp

```diff
@@ -383,3 +385,70 @@ bool AreaTrigger::CheckIsInPolygon2D(Position* pos) const
 
     return locatedInPolygon;
 }
+
+void AreaTrigger::AddAuras(Unit* unit)
+{
+    for (AreaTriggerAuras aura : GetTemplate()->Auras)
+    {
+        if (UnitFitToAuraRequirement(unit, aura.TargetType))
+        {
+            if (Unit* caster = ObjectAccessor::GetUnit(*this, _casterGuid))
+            {
+                caster->AddAura(aura.AuraId, unit);
```

Comment:
some spells are castable while dead, does the spells have limited range that could case too far away issues?

that gameobject cast is bad, don't look on it

AddAura will do nothing for spells with non apply aura spelleffects